client_test.go

```diff
@@ -0,0 +1,377 @@
+package diderot
+
+import (
+	"context"
+	"iter"
+	"log/slog"
+	"maps"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/linkedin/diderot/ads"
+	"github.com/linkedin/diderot/internal/utils"
+	"github.com/linkedin/diderot/testutils"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type Timestamp = timestamppb.Timestamp
+
+var Now = timestamppb.Now
+
+func TestADSClient(t *testing.T) {
+	slog.SetLogLoggerLevel(slog.LevelDebug)
+
+	ts := testutils.NewTestGRPCServer(t)
+
+	server := newMockServer(t)
+	discovery.RegisterAggregatedDiscoveryServiceServer(ts.Server, server)
+	ts.Start()
+
+	client := NewADSClient(ts.Dial(), &ads.Node{Id: "test"})
```

Comment:
pls add a test for responseChunkingSupported being false

Reply:
ACK